### PR TITLE
Initial implementation of hambo avro library

### PR DESCRIPTION
**Pull request by mikecsmith**
Initial ch-kafka-go library build using hambo's avro library - matches API for chs.go/avro

### CODE REVIEW

File: avro/avro.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Package avro provides a user functionality to return the
+// avro encoding of s.
```
Comment (jsimons1989) [inline avro/avro.go]: Can the comment be updated or removed please? What's "encoding of s"?

File: avro/avro.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+package avro
+
+import (
+	"fmt"
+
+	"github.com/hamba/avro"
+)
+
+// Marshaller is an interface for marshalling to and from binary using avro
```
Comment (jsimons1989) [inline avro/avro.go]: We should update these comments to refer to "binary encoded bytes" so it's clear what's happening

File: avro/avro.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+func (marshaller *AvroMarshaller) Marshal(model interface{}) ([]byte, error) {
+	avroSchema, err := avro.Parse(marshaller.Schema)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing avro schema: %s", err)
+	}
+
+	data, err := avro.Marshal(avroSchema, model)
+
+	if err != nil {
+		return nil, fmt.Errorf("error writing avro schema: %s", err)
```
Comment (jsimons1989) [inline avro/avro.go]: This error should be "error marshalling model using Avro schema"

File: avro/avro.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+func (marshaller *AvroMarshaller) Unmarshal(data []byte, model interface{}) error {
+	avroSchema, err := avro.Parse(marshaller.Schema)
+	if err != nil {
+		return fmt.Errorf("error parsing avro schema: %s", err)
+	}
+
+	err = avro.Unmarshal(avroSchema, data, &model)
+	if err != nil {
+		return fmt.Errorf("error reading avro schema: %s", err)
```
Comment (jsimons1989) [inline avro/avro.go]: This error should be "error unmarshalling into model using Avro schema"

File: avro/avro_test.go [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+		So(bufferBytes1, ShouldNotBeNil)
+	})
+
+	Convey("Marshal should return an error unless given a pointer to a struct", t, func() {
+		cs := &AvroMarshaller{
+			Schema: testSchema1,
+		}
+
+		test := "string"
+		bufferBytes1b, err1b := cs.Marshal(test)
```
Comment (jsimons1989) [inline avro/avro_test.go]: Can the variable names be consistent across the tests please?

File: avro/avro.go (hunk 1 above)
Comment (mikecsmith) [reply avro/avro.go]: These are comments left over from the original library in chs.go - don't think they add any value so will remove

File: avro/avro.go (hunk 2 above)
Comment (mikecsmith) [reply avro/avro.go]: done

File: avro/avro.go (hunk 3 above)
Comment (mikecsmith) [reply avro/avro.go]: done

File: avro/avro.go (hunk 4 above)
Comment (mikecsmith) [reply avro/avro.go]: done

File: avro/avro_test.go (hunk 5 above)
Comment (mikecsmith) [reply avro/avro_test.go]: done

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,9 @@
+module github.com/companieshouse/ch-kafka-go
+
+go 1.12
+
+require (
+	github.com/elodina/go-avro v0.0.0-20160406082632-0c8185d9a3ba
```
Comment (jsimons1989) [inline go.mod]: This dependency can be removed

**Review by jsimons1989 — APPROVED** [review]

**Pull request merged**